### DISCUSSION TITLE
Property 'setData' does not exist on type 'Source'

### DISCUSSION THREAD

**Discussion by faraz2135** [maplibre/maplibre-gl-js#1013] [Q&A]
Hello Everyone! I am using "maplibre-gl": "^2.1.1" in my Angular project and I want to animate a point along a line. I am following the example provided in Docs but when I reach this section of code

map.getSource('point').setData(point);

 it says 

"Property 'setData' does not exist on type 'Source'". 

**maplibre-gl-js version**:

**Comment by HarelM** [comment]
`getSource` returns a base `Source` interface.
You need to cast to the relevant source.
Moving this to discussion as this is not an issue.

**Comment by baffioso** [comment]
```ts
(map.getSource('my-source') as GeoJSONSource).setData(...)
```
